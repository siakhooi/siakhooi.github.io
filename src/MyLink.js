import React from 'react'
import PropTypes from 'prop-types'

MyLink.propTypes = {
  desc: PropTypes.string,
  url: PropTypes.string,
  name: PropTypes.string
}

function MyLink (props) {
  let d = props.desc
  if (d !== undefined) d = `- ${d}`
  return (
    <li className="MyLink"><a className="MyLink" href={props.url} rel='nofollow'>{props.name}</a>{d}</li>
  )
}

export default MyLink
