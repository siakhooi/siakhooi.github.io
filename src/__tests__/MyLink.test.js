import React from 'react'
import renderer from 'react-test-renderer'
import MyLink from '../MyLink'

it('MyLink/1', () => {
  const component =
    renderer.create(
      <MyLink
        url={'http://www.google.com'}
        name={'Google'}
      />
    )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

it('MyLink/2', () => {
  const component =
    renderer.create(
      <MyLink
        url={'http://www.google.com'}
        name={'Google'}
        desc={'Search Engine'}
      />
    )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
