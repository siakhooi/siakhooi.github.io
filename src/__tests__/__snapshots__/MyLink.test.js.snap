// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyLink/1 1`] = `
<li
  className="MyLink"
>
  <a
    className="MyLink"
    href="http://www.google.com"
    rel="nofollow"
  >
    Google
  </a>
</li>
`;

exports[`MyLink/2 1`] = `
<li
  className="MyLink"
>
  <a
    className="MyLink"
    href="http://www.google.com"
    rel="nofollow"
  >
    Google
  </a>
  - Search Engine
</li>
`;
